openproject/client.py

```diff
@@ -0,0 +1,96 @@
+import httpx
+from exceptions import APIError, AuthenticationError
+from openproject.types import WorkPackage
+
+
+class Client:
+    def __init__(self, base_url: str, api_token: str):
+        self.base_url = base_url
+        self.api_version = "v3"
+        self.api_token = api_token
+
+    def _handle_response(self, response: httpx.Response):
+        if response.status_code == 401:
+            raise AuthenticationError(response.json()["message"], response.status_code)
+
+        if response.status_code == 204:
+            return None
+
+        try:
+            response.raise_for_status()
+
+            return response.json()
+        except:
+            error = f"{response.text}"
+            content_type = response.headers.get("Content-Type", "").lower()
+            if content_type == "application/json":
+                resp = response.json()
+                error_message_keys = ("message", "msg")
+                error_message = next(
+                    (resp[key] for key in error_message_keys if key in resp), None
+                )
+                if error_message:
+                    error = f"Message: {resp.get('message')} , Error details: {resp.get('errors')}, {resp.get('data')}"
+                    raise APIError(error, response)
```

Comment:
Let's use the extracted `error_message`:

```suggestion
                    error = f"Message: {error_message} , Error details: {resp.get('errors')}, {resp.get('data')}"
                    raise APIError(error, response)
```